File: hbase_storage/src/main/java/com/sap/i076326/client_api.scala
package com.sap.i076326

import java.io.File

class client_api {
  var topicName: String = null
  var tableName: String = null
  var rowName: String = null
  var rowCount: Int = 1
  var columns: String = null
  var location: String = null
  var groupName: String = null

  var producer: com.sap.i076326.kafka_producer = null

  def setTopicName(pTopicName: String): Unit = {
    topicName = pTopicName
  }

  def setZookeeperLocation(pLocation: String): Unit = {
    location = pLocation
  }

  def setConsumerGroup(pGroupName: String): Unit = {
    groupName = pGroupName
  }

  def setTableName(pTableName: String): Unit = {
    tableName = pTableName
  }

  def setRowName(pRowName: String, pRowCount: Int): Unit = {
    rowName = pRowName
    rowCount = pRowCount
  }

  def setColumn(pColumns: String): Unit = {
    columns = pColumns
  }

  def recursiveListFiles(f: File): Array[File] = {
    val these = f.listFiles
    these ++ these.filter(_.isDirectory).flatMap(recursiveListFiles)
  }

  def fetchFiles(directory: String): Unit = {
    val listOfFiles = recursiveListFiles(new File(directory))

    var count = rowCount + 1
    for(fileloc <- listOfFiles){
      val lines = scala.io.Source.fromFile(fileloc.getAbsolutePath()).mkString
      val data = fileloc.toString + ";" + lines

      callKafkaProducer(data, rowName + count)
      count = count + 1
    }
  }

  def initializeProducer(): Unit = {
    producer = new kafka_producer(1, topicName)
  }

  def initializeConsumer(): Unit = {
    val consumer = new kafka_consumer(location, groupName, topicName)
    consumer.run()
  }

  def callKafkaProducer(data: String, row: String): Unit = {
    producer.prepareData(tableName, row, columns, data)
    producer.sendData(producer.createProducerConfig())
  }

  def closeProducer(): Unit = {
    producer.closeProducer()
  }
}

File: hbase_storage_java/src/main/java/com/sap/i076326/client_api.java
package com.sap.i076326;

import java.io.File;
import com.google.common.base.Charsets;
import com.google.common.io.Files;

public class client_api {
    private String topicName;
    private String tableName;
    private String rowName;
    private int rowCount;
    private String columns;

    kafka_producer producer;

    public client_api() {
        this.rowCount = 1;
    }

    public void setTopicName(String topicName) {
        this.topicName = topicName;
    }

    public void setTableName(String tableName) {
        this.tableName = tableName;
    }

    public void setRowName(String rowName, int rowCount) {
        this.rowName = rowName;
        this.rowCount = rowCount;
    }

    public void setColumns(String columns) {
        this.columns = columns;
    }

    public File[] recursiveListFiles(File f) {
        File[] files = f.listFiles();
        return files;
    }

    public void fetchFiles(String directory) {
        File[] listOfFiles = recursiveListFiles(new File(directory));

        int count = rowCount + 1;
        int counter = 0;
        while (counter < listOfFiles.length) {
            String content = "";
            try {
                content = Files.toString(new File(listOfFiles[counter].getAbsolutePath()), Charsets.UTF_8);
            } catch (java.io.IOException e) {

            }
            String data = listOfFiles[counter].getAbsolutePath() + ";" + content;

            callKafkaProducer(data, rowName + count);
            count++;
            counter++;
        }
    }

    public void initializeProducer() {
        producer = new kafka_producer(1, topicName);
    }

    public void callKafkaProducer(String data, String row) {
        producer.prepareData(tableName, row, columns, data);
        producer.sendData(producer.createProducerConfig());
    }

    public void closeProducer() {
        producer.closeProducer();
    }
}